### PR TITLE
Additional features around HA + pre-configuring HEC

### PR DESCRIPTION
**Pull request by rarsan**
- Add support for deploying Splunk tiers across 3 AZs
- Enable multisite clustering by default. Make both indexer and search head clustering AZ-aware
- Enable Http Event Collector (HEC) in indexer tier, with a new load balancer on port 8088. Return HEC url & token as stack outputs.
- Enable indexer discovery so forwarders can retrieve list of indexers dynamically

### CODE REVIEW

File: templates/splunk-enterprise.template [hunk 1]
```diff
@@ -498,6 +605,35 @@
...
+                "VpcId": {
+                    "Ref": "VPCID"
+                },
+                "GroupDescription": "Enable port 8088 on ELB for HEC input",
+                "SecurityGroupIngress": [
+                    {
+                        "IpProtocol": "tcp",
+                        "FromPort": "8088",
+                        "ToPort": "8088",
+                        "CidrIp": "0.0.0.0/0"
```
Comment (sshvans) [inline templates/splunk-enterprise.template]: Do we really need to open this port to world?

Comment (rarsan) [reply templates/splunk-enterprise.template]: HTTP Event Collector endpoint requires a token to accept data, so it's not completely exposed. I agree with you however that it's better not to make public access the default. I'll turn this CIDR block into an additional user parameter.

**Comment by sshvans** [comment]
@rarsan Let me know when you are done from your side and ready to merge the changes.

**Comment by rarsan** [comment]
Thanks for reviewing @sshvans . This is ready to be merged.

**Pull request merged**